### ISSUE TITLE
Incorrect use of moved value reported for copy type

### ISSUE THREAD

**Issue by oribenshir** [intellij-rust/intellij-rust#4361]
## Environment

* **IntelliJ Rust plugin version: 0.2.104.2132-192**
* **Rust toolchain version: nightly-x86_64-pc-windows-gnu 1.38.0**
* **IDE name and version: CLion 2019.2**
* **Operating system: Windows 10**

## Problem description

Inspection falsely identifies a case of use after move. The problem is it report it for built-in types implementing the copy trait, therefore they were never actually moved ('usize' in my case)
Attaching a very simple test case, for code which compiles successfully, but inspection is being reported.

## Steps to reproduce

https://gist.github.com/oribenshir/a05513a114891c8eb1fd8de956e07ecf

**Comment by rrevenantt** [comment]
I had similar issue recently. AFAIR manually cleaning macro expansion cache fixed it (delete `<IdeaConfigDir>/system/rust_expanded_macros` directory, then restart IDE). 

Right now I have accidentally reproduced it again. Just force IDE to update index(File->Invalidate Caches/Restart->Invalidate and Restart). I suppose me and @oribenshir did IDE update recently which caused cache invalidation.

So it looks like `RsMacroExpansionCachesInvalidator` is not working properly for some reason.

**Comment by qiujiangkun** [comment]
Yeah, I have the same issue.
```rust
fn func(pos: usize) {}

fn token(pos: usize) -> Option<usize> {
    func(pos);
    return Some(pos + 1);
}
```
This issue will appear when I write simple codes like this, after the latest update of rust plugin. I am seeking a way to reinstall the previous version.

**Comment by oribenshir** [comment]
> Yeah, I have the same issue.
> 
> ```rust
> fn func(pos: usize) {}
> 
> fn token(pos: usize) -> Option<usize> {
>     func(pos);
>     return Some(pos + 1);
> }
> ```
> 
> This issue will appear when I write simple codes like this, after the latest update of rust plugin. I am seeking a way to reinstall the previous version.

As written above you don't have to. You can clear the cache: https://www.jetbrains.com/help/clion/cleaning-system-cache.html

This did the trick to me aswell

**Comment by qiujiangkun** [comment]
> > Yeah, I have the same issue.
> > ```rust
> > fn func(pos: usize) {}
> > 
> > fn token(pos: usize) -> Option<usize> {
> >     func(pos);
> >     return Some(pos + 1);
> > }
> > ```
> > 
> > 
> > This issue will appear when I write simple codes like this, after the latest update of rust plugin. I am seeking a way to reinstall the previous version.
> 
> As written above you don't have to. You can clear the cache: https://www.jetbrains.com/help/clion/cleaning-system-cache.html
> 
> This did the trick to me aswell
Thanks, problem solved.

**Comment by hlavaatch** [comment]
Just happened to me on CLion 2021.3.3. Rust plugin 0.4.165.4438-213.
Invalidate caches workaround worked for me.

**Comment by Jakobeha** [comment]
This seems to happen after updates (happened once and then again when updating to 0.4.171.4656-221). Regular invalidate caches wasn't enough, but Invalidate caches + Clear file system cache worked.

**Comment by Valentin-Metz** [comment]
This is *still* an issue.
Invalidating caches has not fixed it for me this time.

**Comment by Maksim20023** [comment]
I'm closing this ticket since the problem does not reproduce in the current environment (CLion 2023.2, Rust plugin 0.4.200.5421-232). If this issue still reproduces for you, please reopen the ticket.
